### ISSUE TITLE
Ensure the integration test is sensitive to errors from the stress test repo

### ISSUE THREAD

**Issue by maurolepore** [2DegreesInvesting/PACTA_analysis#186]
@jacobvjk, how can I run the "stress test"? Is this something we we should add to the integration-test.Rmd? Or it's a stand-alone thing that won't be released on the upcoming milestone? 

Relates to:
https://github.com/2DegreesInvesting/PACTA_analysis/pull/183#issuecomment-707589478
https://github.com/2DegreesInvesting/StressTestingModelDev/issues/25

**Comment by jacobvjk** [comment]
It's already being run via source at the very end of script2. 
This is the corresponding script in the StressTest repo https://github.com/2DegreesInvesting/StressTestingModelDev/blob/master/web_tool_stress_test.R . And it is part of the P2020 milestone

**Comment by maurolepore** [comment]
Thanks! Then I need to ensure the integration test breaks when the stress test repo errors. This might be already the case but I need to verify.

**Comment by cjyetman** [comment]
I was wondering if we should add the stress test tool script to [bin/run-r-scripts](https://github.com/2DegreesInvesting/PACTA_analysis/blob/master/bin/run-r-scripts), rather than sourcing it from one of the `web_tool_script_X.R` scripts in this repo. like...
```
Rscript --vanilla web_tool_script_1.R TestPortfolio_Input \
  && Rscript --vanilla web_tool_script_2.R TestPortfolio_Input \
  && Rscript --vanilla web_tool_script_3.R TestPortfolio_Input \
  && Rscript --vanilla ../StressTestingModelDev/web_tool_stress_test.R
```

I don't know if it needs to be run inside of one of the PACTA_analysis scripts... @jacobvjk would know better than I do.

**Comment by jacobvjk** [comment]
@cjyetman I guess it might be possible to extract it, but I do need the project configurations from PACTA_analysis to carry over to the stress test. So if I don't source the stress test from PACTA, I will have to do it the other way round. Unless there is another smarter way to have the project settings carry over, which I am absolutely open for.

**Comment by cjyetman** [comment]
@jacobvjk I doubt there's a good way to do that. That same problem is essentially the root of `create_interactive_report()` having so many arguments... since the repo is separated and it can't rely on anything being in it's environment when it runs, one must explicitly pass everything it needs to it in the arguments

⚖️ ?

**Comment by jacobvjk** [comment]
@cjyetman Okay, but that means that the only other way to make that a bit clearer is to wrap the web version of the stress test in a function in the StressTestingModelDev repo, like you did in create_interactive_report. If this feels better than sourcing it. I am agnostic to that. Or maybe I would prefer it over the current logic, but only do it once the main functionality is finalised.. Does this sound reasonable?

**Comment by cjyetman** [comment]
I would suggest **not** wrapping it in a function... at least not until November. Let's not make anything more difficult for ourselves at the moment. ;-)

**Comment by jacobvjk** [comment]
agreed :)

**Comment by maurolepore** [comment]
> Unless there is another smarter way to have the project settings carry over, which I am absolutely open for.

Not sure if this helps, but one way I can think of making some setting available in the entire R session is via `options()`:

``` r
library(rlang)

some_function <- function(some_option = NULL) {
  some_option <- some_option %||% getOption("some_option")
  
  paste("Do something with a", some_option %||% "default option")
}

getOption("some_option")
#> NULL
some_function()
#> [1] "Do something with a default option"


options(some_option = "custom option")
getOption("some_option")
#> [1] "custom option"
some_function()
#> [1] "Do something with a custom option"
```

<sup>Created on 2020-10-14 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>

**Comment by cjyetman** [comment]
@maurolepore @jacobvjk I believe the discussion here is no longer relevant. Shall we close?

**Comment by cjyetman** [comment]
The current integration tests runs through the full process run on the server in much the same way the server does, including the stress testing and the interactive report making. closing
